File: frontend/src/order/object/core/CreateOrderRequest.ts
import type {
  AddressText,
  CouponId,
  CustomerId,
  IsoDateTime,
  NoteText,
  StoreId,
} from '@/shared/object/domain/DomainObjects'
import type { OrderItemInput } from '@/order/object/core/OrderItemInput'

export type CreateOrderRequest = {
  customerId: CustomerId
  storeId: StoreId
  deliveryAddress: AddressText
  scheduledDeliveryAt: IsoDateTime
  remark?: NoteText
  couponId?: CouponId
  items: OrderItemInput[]
}

File: backend/src/main/scala/domain/order/CreateOrderRequest.scala
package domain.order

import domain.shared.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import domain.shared.*

final case class CreateOrderRequest(
    customerId: CustomerId,
    storeId: StoreId,
    deliveryAddress: AddressText,
    scheduledDeliveryAt: IsoDateTime,
    remark: Option[NoteText],
    couponId: Option[CouponId],
    items: List[OrderItemInput],
)
object CreateOrderRequest:
  given Encoder[CreateOrderRequest] = deriveEncoder
  given Decoder[CreateOrderRequest] = deriveDecoder
